cockatrice/src/tab_server.cpp

```diff
@@ -71,7 +71,11 @@ void RoomSelector::processListRoomsEvent(const Event_ListRooms &event)
     const int roomListSize = event.room_list_size();
     for (int i = 0; i < roomListSize; ++i) {
         const ServerInfo_Room &room = event.room_list(i);
-        
+        QString roomPermissionDisplay = QString::fromStdString(room.privilegelevel()).toLower();
```

Comment:
I _really_ do not like how this block of code here looks. Please add a block comment explaining how this check fully operates. My suggestion on a rewrite (might not be as functional as yours...)
```
/*
 * A room can only have a permission level or a 
 * privilege level. If a room has both, then only
 * the permission level will be checked.
 */
QString roomPermissionDisplay;

if (QString::fromStdString(room.permissionlevel()).toLower() != "none")
	roomPermissionDisplay = QString::fromStdString(room.permissionlevel()).toLower();
else if (QString::fromStdString(room.privilegelevel()).toLower() != "none")
	roomPermissionDisplay = QString::fromStdString(room.privilegelevel()).toLower();

```